broker-daemon/block-order-worker/index.js

```diff
@@ -402,55 +416,69 @@ class BlockOrderWorker extends EventEmitter {
       throw new Error(`No engine available for ${counterSymbol}`)
     }
 
-    const promisedFills = orders.map(async (order) => {
+    const ordersByOrderId = await Promise.all(orders.map((order) => {
+      const range = {
+        gte: order.orderId,
+        lte: order.orderId
+      }
+      return getRecords(this.ordersByOrderId, Order.fromStorage.bind(Order), this.ordersByOrderId.range(range))
+    }))
+
+    const ownOrderIds = ordersByOrderId.filter(order => order && order.length > 0).map(([order]) => order.orderId)
+
+    const promisedFills = []
+    for (let order of orders) {
       const depthRemaining = targetDepth.minus(currentDepth)
 
       // if we have already reached our target depth, create no further fills
       if (depthRemaining.lte(0)) {
         return
       }
 
+      if (ownOrderIds.includes(order.orderId)) {
+        throw new Error(`Cannot fill own order ${order.orderId}`)
```

Comment:
this creates an interesting edge case: if the first order that you want to fill is another user's, but the second is your own, it will fill the first but error out fo the second. Should we handle that separately as part of more general error handling, or should we tackle it now?

Reply:
I think we should leave it for now, but open a ticket specifically outlining this case. It's part of the whole "more nuanced approach to error handling" theme.

Reply:
https://trello.com/c/I08iGste/844-better-error-handling-retrying-when-filling-own-orders